pkg/deploy/migrate-labels.go

```diff
@@ -0,0 +1,103 @@
+//
+// Copyright (c) 2021 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+
+package deploy
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"reflect"
+
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	cheFlavor = os.Getenv("CHE_FLAVOR")
+)
+
+// MigrateCheResourcesLabels searches for objects of different kinds in the cluster,
+// that are added by Che admin and used by the Operator in some way.
+// When such a resource is found, a new label should be added in order to have the resource cached.
+// This should scan all namespaces in order to support all namespaces mode.
+// If an error happens, warning is printed and execution continues.
+// Returns true if everything is updated without errors, false otherwise.
+func MigrateCheResourcesLabels(nonCachingClient client.Client) bool {
+	noErrors := true
+
+	// Prepare selector
+	chePartOfSelectorRequirement, err := labels.NewRequirement(KubernetesPartOfLabelKey, selection.Equals, []string{CheEclipseOrg})
+	if err != nil {
+		logrus.Error("Failed to create selector for resources migration. All custom user objects are lost.")
+		return false
+	}
+	partOfCheLebelSelector := labels.NewSelector().Add(*chePartOfSelectorRequirement)
+	listOptions := &client.ListOptions{
+		LabelSelector: partOfCheLebelSelector,
+	}
+
+	// Migrate all config maps
+	configMapsList := &corev1.ConfigMapList{}
+	err = nonCachingClient.List(context.TODO(), configMapsList, listOptions)
+	if err != nil {
+		logrus.Warn(getFailedToGetErrorMessageFor("Config Maps"))
+		noErrors = false
+	}
+	if configMapsList.Items != nil {
+		for _, cm := range configMapsList.Items {
+			labels := cm.GetLabels()
+			if labels != nil && labels[KubernetesInstanceLabelKey] != cheFlavor {
```

Comment:
I guess you can avoid such `if` using one more label requirement:

labels.NewRequirement(KubernetesInstanceLabelKey, selection.NotEquals, []string{cheFlavor})

Reply:
Technically I can, but to my mind it will make the selector somewhat twisted.

Reply:
Ok, let's improve the selector